vault/audit_broker.go

```diff
@@ -74,19 +94,42 @@ func (a *AuditBroker) Register(name string, b audit.Backend, local bool) error {
 		return fmt.Errorf("%s: name is required: %w", op, event.ErrInvalidParameter)
 	}
 
+	// If the backend is already registered, we cannot re-register it.
+	if a.isRegistered(name) {
+		return fmt.Errorf("%s: backend already registered '%s'", op, name)
+	}
+
+	// Fallback devices are singleton instances, we cannot register more than one or overwrite the existing one.
+	if b.IsFallback() && a.fallbackBroker.IsAnyPipelineRegistered(eventlogger.EventType(event.AuditType.String())) {
+		existing, err := a.existingFallbackName()
```

Comment:
Since an error is inevitable in this condition (when both `b.IsFallback()` and `a.fallbackBroker.IsAnyPipelineRegistered(eventlogger.EventType(event.AuditType.String()))` are `true`), why not have the **existingFallbackName** function simply return an error already properly formatted?

Reply:
Sorry, I forgot the emoji codes, this concern is non-blocking.

Reply:
I see what you mean. The idea when writing the function was that it could be reusable. In hindsight, maybe that was a bit 🔮,  but I find that outside of context of this one call `existingFallbackName` makes more sense as a code unit this way. Basically, I agree with you, but at this point I don't really want to change it since I don't see much of a difference either way 🤔